contracts/utils/Pausable.sol

```diff
@@ -36,7 +36,7 @@ abstract contract Pausable is Context {
     /**
      * @dev Returns true if the contract is paused, and false otherwise.
      */
-    function paused() public view returns (bool) {
+    function paused() public view virtual returns (bool) {
```

Comment:
The issue with making this virtual is that the Paused/Unpaused events stop being reliable. What do you think?

Reply:
Thats is true of all virtual functions that trigger events. Following this argument, `ERC20._transfer` should not be made virtual.

I think that overriding (non-hook) functions requiers a good understanding of the contract being overridden.

This question is more about the usage we expect from pausable:
- should pausable be a mechnism that expects explicit calls to the `_pause` & `_unpause` functions, which trigger event, so that the pausable status can be follow from events. This is the logic we want if we see pausable a a tool for defender.
- should pausable be a building blocks that provide modifier that can be used in your contract and are compatible with any pausing/unpausing pattern. There is a default one (relying on storage, events, and internal functions), but this is just an example and this can be changed to any logic (like a "pause on sunday" logic) that will not necessarily trigger events / be compatible with defender.

I prefer the second option, but that is just my point of view

Reply:
>Following this argument, ERC20._transfer should not be made virtual.

Why not? In ERC20 it's not possible to make a transfer without emitting the event, even if `_transfer` is virtual.

But the argument does apply to not making `ERC20.balanceOf` virtual. And `ERC721.ownerOf`... But that would be basically against the spec so people would know not to do it.

 `Pausable` is not specified, so we would have to document the semantics. But it feels impossible to override `paused` in a way that preserves the semantics that events fully track pausing and unpausing.

Reply:
> Pausable is not specified, so we would have to document the semantics. But it feels impossible to override paused in a way that preserves the semantics that events fully track pausing and unpausing.

True. I just think that the semantics are not essential in this case (not an ERC?) and the intent to pause function through a modifier is the key element, regardless of changes being notified by events.
The change should be notified when "triggered", but I don't have an issue with pausable patterns where the change is not "triggered", but a consequence of an external element (time, other contract)